BeerCatalogFullstack/FrontEnd/src/components/ProfilePage/ProfileSection/profileSection.jsx

```diff
@@ -0,0 +1,150 @@
+import React from 'react';
+import moment from 'moment';
+import { withRouter } from 'react-router-dom';
+import SimpleReactValidator from 'simple-react-validator';
+import PropTypes from 'prop-types';
+
+import Input from 'components/common/Input/input';
+import PhotoSelector from 'components/common/PhotoSelector/photoSelector';
+import UserPreferenceSection from 'components/profilePage/UserPreferenceSection/userPreferenceSection';
+import profileValidationConfig from 'validationConfigs/profileValidationConfig';
+import userService from 'services/userService';
+import serviceWrapper from 'helpers/serviceWrapper';
+import redirectHelper from 'helpers/redirectHelper';
+import { UserContext } from 'store/context/userContext';
+
+import './profileSection.scss';
+
+class ProfileSection extends React.PureComponent {
+    static propTypes = {
+        history: PropTypes.shape({
+            length: PropTypes.number.isRequired,
+            action: PropTypes.string.isRequired,
+            location: PropTypes.shape({
+                pathname: PropTypes.string.isRequired,
+                search: PropTypes.string.isRequired,
+                hash: PropTypes.string.isRequired,
+                key: PropTypes.string.isRequired
+            }),
+            push: PropTypes.func.isRequired
+        }).isRequired
+    }
+
+    static contextType = UserContext;
+
+    constructor(props) {
+        super(props);
+        this.validator = new SimpleReactValidator();
+        this.errorFieldRef = React.createRef();
+        this.state = {
+            photo: '',
+            name: '',
+            email: '',
+            birthdate: ''
+        };
+    }
+
+    async componentDidMount() {
+        const user = await userService.getUser(this.context.userId);
+        this.setState({
+            id: user.id,
+            name: user.name,
+            email: user.email,
+            photo: user.photo,
+            birthdate: user.birthdate ? moment(user.birthdate).format('YYYY-MM-DD') : ''
+        });
+    }
+
+    deletePhoto = () => {
+        this.setState({ photo: '' });
+    }
+
+    changeName = (e) => {
+        this.setState({ name: e.target.value });
+    }
+
+    changeEmail = (e) => {
+        this.setState({ email: e.target.value });
+    }
+
+    changeBirthdate = (e) => {
+        this.setState({ birthdate: moment(e.target.value).format('YYYY-MM-DD') });
```

Comment:
Since you use this logic several times, please create a helper with a method like `getDate`